tools/clang/test/CodeGenSPIRV/spirv.opt.gis.hlsl

```diff
@@ -1,5 +1,12 @@
-// RUN: not %dxc -T ps_6_0 -E main -spirv -Gis %s 2>&1 | FileCheck %s
+// RUN: %dxc -T ps_6_0 -E main -spirv -Gis %s 2>&1 | FileCheck %s
 
-void main() {}
+// Make sure the no-contraction is added for the arithmetic operation.
+// CHECK: OpDecorate [[op:%[0-9]+]] NoContraction
+
+float4 v;
+float4 main(uint col : COLOR) : SV_Target0
+{
+  // [[op]] = OpVectorTimesScalar
+  return 3*v;
```

Comment:
Shouldn't several operation types/basetypes be tested?

Reply:
I thought about that. I'm not sure how many too do. This was meant to test whether the option is read and used correctly. Adding more types and operations is testing if `isArithmeticInstruction` works correctly, which gets covered by the decoration.no-contraction*.hlsl tests.

Let me know what you think I need to add if you do not the existing tests are good enough.

Reply:
I agree. I was wondering if those test wouldn't catch changes in our spaghetti codegen shortcuts, but since this change only impacts the InitInstr function, seems fine.